### PR TITLE
Auto tests: log cluster errors and save init script logs as build artifacts

### PR DESCRIPTION
**Pull request by rtitle**
This should help make automation tests easier to debug.

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: automation/project/Dependencies.scala [hunk 1]
```diff
@@ -6,7 +6,7 @@ object Dependencies {
   val akkaHttpV = "10.0.5"
 
   val workbenchModelV   = "0.10-6800f3a"
-  val workbenchGoogleV  = "0.15-2fc79a3"
+  val workbenchGoogleV  = "0.16-2947b3a-SNAP"
```
Comment (rtitle) [inline automation/project/Dependencies.scala]: For https://github.com/broadinstitute/workbench-libs/pull/94

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala [hunk 2]
```diff
@@ -81,7 +86,15 @@ trait LeonardoTestUtils extends WebBrowserSpec with Matchers with Eventually wit
                    expectedStatuses: Iterable[ClusterStatus],
                    clusterRequest: ClusterRequest): Cluster = {
 
-    expectedStatuses should contain (cluster.status)
+    // Always log cluster errors
+    if (cluster.errors.nonEmpty) {
+      logger.warn(s"Cluster ${cluster.googleProject}/${cluster.clusterName} returned the following errors: ${cluster.errors}")
+    }
+
+    withClue(s"Cluster ${cluster.googleProject}/${cluster.clusterName}: ") {
```
Comment (rtitle) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: Error messages like:
```
List(Running, Error) did not contain element Creating
```
become: 
```
Cluster gpalloc-qa-master-6jj5168/automation-test-abdndktwz: List(Running, Error) did not contain element Creating
```
See http://doc.scalatest.org/1.8/org/scalatest/AppendedClues.html

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala [hunk 3]
```diff
@@ -101,15 +114,27 @@ trait LeonardoTestUtils extends WebBrowserSpec with Matchers with Eventually wit
...
 
     // verify with get()
-    clusterCheck(Leonardo.cluster.get(googleProject, clusterName), googleProject, clusterName, Seq(ClusterStatus.Creating), clusterRequest)
+    val creatingCluster = clusterCheck(Leonardo.cluster.get(googleProject, clusterName), googleProject, clusterName, Seq(ClusterStatus.Creating), clusterRequest)
 
     // wait for "Running" or error (fail fast)
     implicit val patienceConfig: PatienceConfig = clusterPatience
-    val actualCluster = eventually {
-      clusterCheck(Leonardo.cluster.get(googleProject, clusterName), googleProject, clusterName, Seq(ClusterStatus.Running, ClusterStatus.Error), clusterRequest)
+    val actualCluster = Try {
```
Comment (rtitle) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: Wrapped in `Try` so even if this fails we download the init bucket logs

**Comment by rtitle** [comment]
jenkins retest

**Comment by rtitle** [comment]
jenkins retest

**Comment by rtitle** [comment]
jenkins retest

**Comment by rtitle** [comment]
jenkins retest

**Comment by rtitle** [comment]
jenkins retest

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala (hunk 2 above)
Comment (aweng98) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: 👍

Comment (helgridly) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: niiiiiice

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala (hunk 3 above)
Comment (helgridly) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: so the check failing will fail the test but still return a Failure of... some kind of test-failed-exception???

Comment (rtitle) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: We call `.get` on the `Try` below which will re-throw the exception. I just wanted to make `saveClusterInitLogFile` run regardless.

**Comment by rtitle** [comment]
Thanks. It worked locally, gonna double check in this test run though.

**Comment by rtitle** [comment]
Damn, it saved the file:
```
Saved log file for cluster gpalloc-qa-master-6jj5168/ClusterName(automation-test-a8zvoms3z) to /app/cluster-logs/automation-test-a8zvoms3z--ebf30f03-4e19-41e1-a9b6-da6000d22a98/2018-03-07T15:19:24.739Z-dataproc-initialization-script-0_output.log
```
but it didn't show up as a Jenkins build artifact: https://fc-jenkins.dsp-techops.broadinstitute.org/job/leonardo-fiab-test-runner/518/

Looking into it.

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala [hunk 4]
```diff
@@ -82,7 +87,15 @@ trait LeonardoTestUtils extends WebBrowserSpec with Matchers with Eventually wit
                    expectedStatuses: Iterable[ClusterStatus],
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: `Status` or `Statuses`?

Comment (aweng98) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: surprise to me, googled this. `status and statuses can both be used as the plural, depending on the context`

Comment (rtitle) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: I've heard `stati` too.
I personally use `statuses`, others may disagree. :)

Comment (helgridly) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: etymology of status is Latin, so stati is plausible, except that we're not dead Romans so their grammar rules are irrelevant to us

**Comment by rtitle** [comment]
This now works:
image

**Comment by rtitle** [comment]
jenkins retest

**Pull request merged**